src/maliput/routing/route.cc

```diff
@@ -360,5 +365,130 @@ int Route::FindDirectionTowardsLaneSRangeWithStraightPredecessor(const Route::La
+      api::IsInertialPositionClose(inertial_pos_a, inertial_pos_b, tolerance);
+  return !comparison_result.message.has_value();
+}
+
+// Returns the closes api::LaneEnd::Which of the api::Lane that @p position is in.
+api::LaneEnd::Which GetClosestLaneEndWhich(const api::RoadPosition& position) {
+  const api::Lane* lane = position.lane;
+  return position.pos.s() <= lane->length() - position.pos.s() ? api::LaneEnd::Which::kStart
+                                                               : api::LaneEnd::Which::kFinish;
+}
+
+// Matches @p position_a in the preceeding Phase with an api::RoadPosition in @p positions_b from
+// the suceeding Phase.
+// Filters @p positions_b using @p ongoing_lane_end_set to look for valid positions in the topological
+// sense to then find the first match in the subset by geometrical correspondence within @p tolernace.
+// @param position_a api::RoadPosition at the end of the preceeding Phase.
+// @param positions_b api::RoadPositions at the start of the suceeding Phase.
+// @param ongoing_lane_end_set The ongoing api::LaneEndSet that corresponds with @p position_a.
+// @param tolerance Tolerance to compare the positions.
+// @return An optional with an api::RoadPosition from @p positions_b that corresponds with @p position_a.
+std::optional<api::RoadPosition> MatchPositionInLaneEndSet(const api::RoadPosition& position_a,
+                                                           const std::vector<api::RoadPosition> positions_b,
+                                                           const api::LaneEndSet& ongoing_lane_end_set,
+                                                           double tolerance) {
+  std::vector<api::RoadPosition> matching_positions_b{};
+  std::copy_if(positions_b.begin(), positions_b.end(), std::back_inserter(matching_positions_b),
+               [&ongoing_lane_end_set](const api::RoadPosition& position) {
+                 for (int i = 0; i < ongoing_lane_end_set.size(); ++i) {
+                   if (position.lane == ongoing_lane_end_set.get(i).lane) {
+                     return true;
+                   }
+                 }
+                 return false;
+               });
+  const auto it = std::find_if(matching_positions_b.begin(), matching_positions_b.end(),
+                               [position_a, tolerance](const api::RoadPosition& position) {
+                                 return IsRoadPositionClose(position_a, position, tolerance);
+                               });
+  return it != matching_positions_b.end() ? std::optional<api::RoadPosition>{*it} : std::nullopt;
+}
+
+}  // namespace
+
+std::vector<std::string> Route::ValidateEndToEndConnectivity() const {
+  std::vector<std::string> errors;
+
+  // The start Phase must have only one start position.
+  if (phases_.front().start_positions().size() != 1u) {
+    errors.push_back("Start Phase::start_positions() does not have one position.");
+  }
+  // The end Phase must have only one start position.
```

Comment:
"start" should be "end".

Reply:
[5f99c1c](https://github.com/maliput/maliput/pull/628/commits/5f99c1c613ff4ae3206bf0796b0287698f0d08ae)